pyps4/ddp.py

```diff
@@ -66,18 +68,36 @@ def get_ddp_launch_message(credential):
 
 def _send_recv_msg(host, broadcast, msg, receive=True):
     """Send a ddp message and receive the response."""
-    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    sock.bind((UDP_IP, UDP_PORT))
-    sock.settimeout(3.0)
 
-    if broadcast:
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-        host = '255.255.255.255'
-
-    sock.sendto(msg.encode('utf-8'), (host, DDP_PORT))
-
-    if receive:
-        return sock.recvfrom(1024)
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    except OSError as error:
+        _LOGGER.error('failed to create socket, %s', error)
+        return
+    try:
+        sock.bind((UDP_IP, UDP_PORT))
+        sock.settimeout(3.0)
+    except OSError as error:
+        _LOGGER.error('failed to bind socket %s:%s, %s', UDP_IP, UDP_PORT, error)
+        sock.close()
+        return
+
+    try:
+        if broadcast:
```

Comment:
this should be removed form the try, except block ... maybe in another

Reply:
what part are you referring to here?